README.md

```diff
@@ -1,41 +1,158 @@
+<h2 id="installation">Installation</h2>
 
+To run Habitat, you need:
+- [Python 3.6+](https://www.python.org/)
+- [Pytorch 1.1.0+](https://pytorch.org/)
+- A system equiped with an Nvidia GPU.
 
-## License
+Currently, we have predictors for the following Nvidia GPUs:
+
+| GPU        | Generation  | Memory | Mem. Type | SMs |
+| ---------- |:-----------:| ------:| :-------: | :-: |
+| P4000      | Pascal      | 8 GB   | GDDR5     | 14  |
+| P100       | Pascal      | 16 GB  | HBM2      | 56  |
+| V100       | Volta       | 16 GB  | HBM2      | 80  |
+| 2070       | Turing      | 8 GB   | GDDR6     | 36  |
+| 2080Ti     | Turing      | 11 GB  | GDDR6     | 68  |
+| T4         | Turing      | 16 GB  | GDDR6     | 40  |
+| 3090       | Ampere      | 24 GB  | GDDR6X    | 82  |
+
+**NOTE:** Not implmented yet
+```zsh
+python3 -m pip install habitat
+python3 -c "import habitat"
+```
+
+<h2 id="build">Building from source</h2>
+
+Prerequsites:
+- A system equiped with an Nvidia GPU with properly configured CUDA
+- [CUDA Toolkit](https://developer.nvidia.com/cuda-toolkit-archive)
+- [cmake v3.17+](https://github.com/Kitware/CMake/releases)
+- [Habitat pre-trained models](https://zenodo.org/record/4876277)
+
+```zsh
+git clone https://github.com/CentML/habitat.git && cd habitat
+git submodule init && git submodule update
+
+# Download the pre-trained models
+cd analyzer
+curl -O https://zenodo.org/record/4876277/files/habitat-models.tar.gz\?download\=1
+
+# Install the models
+./extract-models.sh
+```
+
+**Note:** Habitat needs access to your GPU's performance counters, which requires special permissions if you are running with a recent driver (418.43 or later). If you encounter a `CUPTI_ERROR_INSUFFICIENT_PRIVILEGES` error when running Habitat, please follow the instructions [here](https://developer.nvidia.com/ERR_NVGPUCTRPERM) and in [issue #5](https://github.com/geoffxy/habitat/issues/5).
+
+### Building with Docker
+
+1. Run `setup.sh` under `docker/` to build the Habitat container image.
+1. Run `start.sh` to start a new container. By default, your home directory will be mounted inside the container under `~/home`.
+1. Once inside the container, run `install-dev.sh` under `analyzer/` to build and install the Habitat package.
+1. In your scripts, `import habitat` to get access to Habitat. See `experiments/run_experiment.py` for an example showing how to use Habitat.
+
+### Building without Docker
+
+```zsh
+# Sanity check, following command should return without errors
+nvidia-smi
+
+# Install CUPTI
+# Find approriate installer for your version of CUDA from:
+# https://developer.nvidia.com/cuda-toolkit-archive
+# For example, the install binary for CUPTI for 11.7.1
+wget https://developer.download.nvidia.com/compute/cuda/11.7.1/local_installers/cuda_11.7.1_515.65.01_linux.run
+
+chmod +x cuda_11.7.1_515.65.01_linux.run
+
+# Make sure to only install the toolkit
+./cuda_11.7.1_515.65.01_linux.run
```

Comment:
Do you have to install CUPTI from the runfile? Is CUPTI not available from [rmp or deb](https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=22.04&target_type=deb_network)? Similarly for cmake?
I just feel like the users could get lost easily with such a long installation instruction. From a user experience point of view, it'd be better if we have a list of scripts and a multiplexer to select between them depending on the user's distro. E.g., for Ubuntu 22.02, run this command. For RedHat, run that command, etc. And we don't need to support that many distros. I personally feel like Ubuntu is 90% of the Linux for DL market right now.